### PR TITLE
IPA Provisioning and Updated Manual

### PR DESCRIPTION
**Pull request by ankitabhopatkar13**

### CODE REVIEW

File: build_binary.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -e
+
+yum install rpm-build rpmdevtools python-devel -y
+yum groupinstall "Development Tools" -y
+
+rpmdev-setuptree
+
+pip install git+https://github.com/thoughtworksinc/pex#egg=pex
```
Comment (andreikeis) [inline build_binary.sh]: why does it point to toughtworks repo?

File: treadmill/infra/constants.py [hunk 2]
```diff
@@ -20,4 +20,12 @@
...
 }
+ROLES = {
+    'MASTER': 'MASTER',
+    'NODE': 'NODE',
+    'LDAP': 'LDAP',
+    'IPA': 'IPA',
+    'ZOOKEEPER': 'ZOOKEEPER',
+    'DEFAULT': 'DEFAULT'
+}
 DEFAULT_DOMAIN = 'treadmill.org'
```
Comment (andreikeis) [inline treadmill/infra/constants.py]: on a second thought, i would make it required argument and not have it in constants. it really does not belong, easy to specify (I doubt it is correct value for anyone) and we do not really own treadmill.org unfortunately.

File: build_binary.sh (hunk 1 above)
Comment (snyaggarwal) [inline build_binary.sh]: We had to patch PEX sometime back, to persist permissions on files bundled inside PEX. 
We have raised the pull request to PEX to merge this change, its on their list. Till then we have to use this patched PEX.
Can move this to MS repo?

File: treadmill/infra/constants.py (hunk 2 above)
Comment (snyaggarwal) [inline treadmill/infra/constants.py]: Done!

File: build_binary.sh (hunk 1 above)
Comment (andreikeis) [inline build_binary.sh]: probably separate pr.

**Pull request merged**